src/boot/efi/devicetree.c

```diff
@@ -72,8 +72,6 @@ EFI_STATUS devicetree_install(struct devicetree_state *state, EFI_FILE *root_dir
         assert(name);
 
         state->orig = find_configuration_table(MAKE_GUID_PTR(EFI_DTB_TABLE));
-        if (!state->orig)
```

Comment:
This check should stay, and a debug level log be written, instead of returning (same below)

Reply:
If there can't be a debug message here than maybe no message is best.

We were told to install a devicetree and we installed one. I'm not sure that is a need for more "drama" if we install a devicetree on a system without a DT than a system with an existing DT.

Reply:
+1 for this, installing a DT should not implicitly depend on whatever environment the EFI firmware provides, it's intentional - and if not then usually a pretty obvious thing to spot.